### PR TITLE
Added telelmetry for settings v2 and v1

### PR DESCRIPTION
**Pull request by srinathnarayanan**
- Modified tab activate and close logging to contain tabId
- Overrode this tabId in SettingsV2 to make the tabId contain "SettingsV2"
- Changed type of telemetry action for updatecollection and update offer success and failure to "UpdateSettingsV2" in SettingsComponent
- Added new Action types "DiscardSettings" and "DiscardSettingsV2" for telemetry actions fired when discard button is clicked on setttings v2 and settings v2 respectively

### CODE REVIEW

File: src/Shared/Telemetry/TelemetryConstants.ts [hunk 1]
```diff
@@ -86,7 +86,10 @@ export enum Action {
   CreateMongoCollectionWithWildcardIndex,
   ClickCommandBarButton,
   RefreshResourceTreeMyNotebooks,
-  ClickResourceTreeNodeContextMenuItem
+  ClickResourceTreeNodeContextMenuItem,
+  DiscardSettings,
+  UpdateSettingsV2,
+  DiscardSettingsV2
```
Comment (swviswan) [inline src/Shared/Telemetry/TelemetryConstants.ts:92]: >DiscardSettingsV2 [](start = 2, length = 17)

uggest: prefix with area like: SettingsUpdated, SettingsV2Updated, NotebooksRunCell...etc. So we can find all telemetry constants related to an area together when writing them in sorted order in this list.

**Review by swviswan — APPROVED** [review]
:shipit:

File: src/Shared/Telemetry/TelemetryConstants.ts (hunk 1 above)
Comment (srinathnarayanan) [reply src/Shared/Telemetry/TelemetryConstants.ts:92]: the old action for updating settings was "UpdateSettings", so I followed that pattern. but, sure, will change it for SettingsV2

File: src/Explorer/Tabs/SettingsTab.ts [hunk 2]
```diff
@@ -1233,6 +1233,9 @@ export default class SettingsTab extends TabsBase implements ViewModels.WaitsFor
   };
 
   public onRevertClick = (): Q.Promise<any> => {
+    TelemetryProcessor.trace(Action.DiscardSettings, ActionModifiers.Mark, {
+      message: "Settings Discarded"
```
Comment (languy) [inline src/Explorer/Tabs/SettingsTab.ts:1237]: Not sure whether this is needed, since the Action already describes. We can save a few bytes.

**Review by languy — APPROVED** [review]

**Pull request merged**